app/controllers/posts_controller.rb

```diff
@@ -29,13 +29,11 @@ def update
   end
 
   def index
-    @posts = Post.all
+    @posts = Post.published
     if params[:query].present?
-      @posts = Post.search(params[:query]).order('views_count DESC , created_at DESC')
-    else
-      @posts = Post.published.order('views_count DESC , created_at DESC')
+      @posts = @posts.search(params[:query])
```

Comment:
Yes you would need to reassign the results of @posts.search back to @posts. Such method calls does not change the original object the method is called on. It just returns the value.